app/src/main/java/com/example/cristiana/weatherapp/TenDaysActivity.java

```diff
@@ -0,0 +1,212 @@
+
+        Call<TenDaysWeather> callable = WeatherApp.Service.Get().getTenDaysWeather(mLocationName, units, numDays, key);
+
+        callable.enqueue(new Callback<TenDaysWeather>() {
+            @Override
+            public void onResponse(Call<TenDaysWeather> call, Response<TenDaysWeather> response) {
+                if (response.isSuccessful()) {
+                    TenDaysWeather tenDaysWeather = response.body();
+                    adapter.setData(tenDaysWeather);
+                    adapter.notifyDataSetChanged();
+                } else {
+                    Toast.makeText(TenDaysActivity.this, "onResponse was unsuccessful", Toast.LENGTH_SHORT).show();
+                }
+            }
+
+            @Override
+            public void onFailure(Call<TenDaysWeather> call, Throwable t) {
+                t.printStackTrace();
+                Toast.makeText(TenDaysActivity.this, "onFailure", Toast.LENGTH_SHORT).show();
+            }
+        });
+    }
+
+    static class Adapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+        List<Forecast> mForecast = new ArrayList<>();
+        Callback mCallback;
+
+        public Adapter(Callback callback) {
+            mCallback = callback;
+        }
+
+        public interface Callback {
+            void show (Forecast forecast);
+        }
+
+        public void setData(TenDaysWeather tenDaysWeather) {
+            List<Forecast> forecasts = tenDaysWeather.getForecast();
+            for (Forecast forecast : forecasts) {
+                mForecast.add(forecast);
+            }
+        }
+
+        @Override
+        public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_day, parent, false);
+            return new ViewHolder(view);
+        }
+
+        @Override
+        public void onBindViewHolder(RecyclerView.ViewHolder holder, final int position) {
+            ((ViewHolder) holder).bind(mForecast.get(position), new View.OnClickListener() {
+
+                @Override
+                public void onClick(View v) {
+                    mCallback.show(mForecast.get(position));
+                }
+            });
+        }
+
+        @Override
+        public int getItemCount() {
+            return mForecast.size();
+        }
+
+        static class ViewHolder extends RecyclerView.ViewHolder {
+            private final TextView mCrtDate;
+            private final TextView mCrtTemperature;
+
+            public ViewHolder(View itemView) {
+                super(itemView);
+
+                mCrtDate = (TextView) itemView.findViewById(R.id.crtDay);
+                mCrtTemperature = (TextView) itemView.findViewById(R.id.crtTemperature);
+            }
+
+            public void bind(Forecast forecast, View.OnClickListener onClickListener) {
+                String[] months = {"January", "February", "March", "April", "May", "June", "July",
```

Comment:
Use a SimpleDateFormat for formatting the full month name. It will save you the hassle of having to keep track of month strings